File: SchoolWebApp-API/Project.Core/Entities/Students/StudentParent.cs
using SchoolWebApp.Core.Entities.Settings;
using SchoolWebApp.Core.Entities.Shared;
using System.ComponentModel.DataAnnotations.Schema;

namespace SchoolWebApp.Core.Entities.Students
{
    public class StudentParent: Base
    {
        [NotMapped]
        public override int Id { get; set; }
        public int RelationShipId { get; set; }
        public RelationShip? RelationShip { get; set; }
        public int StudentId { get; set; }
        public Student? Student { get; set; }
        public int ParentId { get; set; }
        public Parent? Parent { get; set; }
        public string? OtherDetails { get; set; }
    }
}

File: SchoolWebApp-Client/src/app/students/models/student-parent.ts
import { ResourceModel } from "@/core/models/ResourceModel";
import { StudentDetails } from "./student-details";
import { Relationship } from "@/settings/models/relationship";
import { Parent } from "./parent";

export class StudentParent extends ResourceModel<StudentParent> {
    public OtherDetails?: string;

    public studentId?: number;
    public student?: StudentDetails;

    public parentId?: number;
    public parent?: Parent;

    public relationShipId?: number;
    public relationShip?: Relationship;

    constructor(model?: Partial<StudentParent>) {
        super(model);
    }
}